docs/guides/docs/intro-to-sway/rust-sdk.mdx

```diff
@@ -0,0 +1,323 @@
+---
+title: Rust Testing
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Testing the contract
+
+You can find the full contract code, along with example tests using the Rust SDK, in the provided repository.
+
+### Generating a Test Template in Rust
+
+To create your own test template using Rust, follow these steps with `cargo-generate`:
+
+1. Install `cargo-generate`:
+```bash
+cargo install cargo-generate
+```
+
+2. Generate the template:
+```bash
+cargo generate --init fuellabs/sway templates/sway-test-rs --name contract
+```
+
+### Running the Tests
+
+For running tests located in `harness.rs`, use:
+```bash
+cargo test
+```
+
+If you wish to print outputs to the console during tests, execute:
+```bash
+cargo test -- --nocapture
+```
+
+## Imports
+
+At the beginning of the harness.rs file, import the Fuel Rust SDK. The majority of the functionalities you'll need are housed within the prelude.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_import"
+  commentType="//"
+  lang="rust"
+/>
+
+Always compile your contracts after making any changes. This ensures you're working with the most recent `contract-abi` that gets generated.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_abi"
+  commentType="//"
+  lang="rust"
+/>
+
+## Initializing Functions
+
+When writing tests for Sway, two crucial objects are required: the contract instance and the wallets that interact with it. This helper function ensures a fresh start for every new test case. It will export the deployed contracts, the contract ID, and all the generated wallets for this purpose.
+
+<CodeImport
```

Comment:
It's unclear if we are supposed to copy and paste this or not. It's different that what is generated. I would take a look at the Fuel Journey Rust guide to see if that helps with outlining this.

Reply:
This includes the changes from rust sdk v0.48.0 so its an updated version of what is generated. I added a note in there to help clear up that the user needs to copy this in.